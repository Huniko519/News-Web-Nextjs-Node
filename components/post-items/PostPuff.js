import React from 'react';
import PropTypes from 'prop-types';
import PostItemBadge from './PostItemBadge';
import PostItemMedia from './PostItemMedia';
import PostItemTitle from './PostItemTitle';
import PostItemLabel from './PostItemLabel';
import PostItemPubDate from './PostItemPubDate';
import ScopedColorContext from '../ScopedColorContext';

/**
 * Post puffs feature a square image on the left with the post headline and primary category
 * to the right, these are arranged in a classic  media object layout such as below.
 *
 * ╔═══════╤══════════════╗
 * ║       │   Category   ║
 * ║ Image │              ║
 * ║       │   Headline   ║
 * ╚═══════╧══════════════╝
 *
 * @param {post: {...postData}} props
 */
const PostPuff = (props) => {
  const {
    scoped, post, showAuthor, showPubDate,
  } = props;
  const colorKeys = post.breadcrumbs.map((breadcrumb) => breadcrumb.slug).reverse();

  return (

    <div className={`inews__post inews__post-puff ${showAuthor ? 'show-author' : ''}`} key={post.id} data-post-id={post.id}>
      <PostItemMedia
        className="inews__post-puff__media"
        imageSize="small-1:1"
        post={post}
        height={155}
        width={155}
        layout="fixed"
        showAuthor={showAuthor}
        placeholder="/static/images/placeholder/placeholder-84x84.png"
      />

      <div className="inews__post-puff__content">
        <div>

          {
            scoped
              ? (
                <ScopedColorContext colorKeys={colorKeys}>
                  <PostItemBadge post={post} showAuthor={showAuthor} />
                </ScopedColorContext>
              )
              : <PostItemBadge post={post} showAuthor={showAuthor} />
          }

          <div className="inews__post-puff__content-headline">
            <PostItemLabel post={post} />
            <PostItemTitle post={post} />
          </div>
        </div>
        <PostItemPubDate post={post} showPubDate={showPubDate} />
      </div>
    </div>
  );
};

/* istanbul ignore next */
PostPuff.propTypes = {
  scoped: PropTypes.bool,
  post: PropTypes.objectOf(PropTypes.any).isRequired,
  showAuthor: PropTypes.bool,
  showPubDate: PropTypes.bool,
};

/* istanbul ignore next */
PostPuff.defaultProps = {
  scoped: false,
  showAuthor: false,
  showPubDate: false,
};

export default PostPuff;
