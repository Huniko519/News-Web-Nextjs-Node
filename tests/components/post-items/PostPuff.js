import React from 'react';
import PostPuff from '../../../components/post-items/PostPuff';

beforeEach(() => {
  jest.clearAllMocks();
});

test('renders the snapshot', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postPuff = shallow(<PostPuff post={post} />);
  expect(postPuff).toMatchSnapshot();
});

it('should output with the correct class on the root element', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postPuff = shallow(<PostPuff post={post} />);
  expect(postPuff.hasClass('inews__post')).toBe(true);
  expect(postPuff.hasClass('inews__post-puff')).toBe(true);
});

it('should call the <PostItemMedia /> component with the correct props', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postPuff = shallow(<PostPuff post={post} />);
  const postItemMedia = postPuff.find('PostItemMedia');
  expect(postItemMedia.length).toBe(1);
  expect(postItemMedia.prop('className')).toBe('inews__post-puff__media');
  expect(postItemMedia.prop('imageSize')).toBe('small-1:1');
  expect(postItemMedia.prop('showAuthor')).toBe(false);
  expect(postItemMedia.prop('post')).toMatchObject(post);
});

it('should call the <PostItemTitle /> component with the correct props', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postPuff = shallow(<PostPuff post={post} />);
  const postItemMedia = postPuff.find('PostItemTitle');
  expect(postItemMedia.length).toBe(1);
  expect(postItemMedia.prop('post')).toMatchObject(post);
});

it('should output a <PostItemBadge /> component', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postPuff = shallow(<PostPuff post={post} />);
  expect(postPuff.find('PostItemBadge').length).toBe(1);
  expect(postPuff.find('PostItemBadge').prop('post')).toMatchObject(post);
});
