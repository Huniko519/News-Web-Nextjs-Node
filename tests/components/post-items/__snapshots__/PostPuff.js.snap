// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the snapshot 1`] = `
<div
  className="inews__post inews__post-puff "
  data-post-id={1}
  key="1"
>
  <PostItemMedia
    className="inews__post-puff__media"
    height={155}
    imageSize="small-1:1"
    layout="fixed"
    linkToAuthor={false}
    placeholder="/static/images/placeholder/placeholder-84x84.png"
    post={
      Object {
        "breadcrumbs": Array [
          Object {
            "name": "News",
          },
        ],
        "id": 1,
        "link": "https://inews-article.com/test-article",
        "title": Object {
          "rendered": "title",
        },
      }
    }
    postItemType=""
    showAuthor={false}
    width={155}
  />
  <div
    className="inews__post-puff__content"
  >
    <div>
      <PostItemBadge
        post={
          Object {
            "breadcrumbs": Array [
              Object {
                "name": "News",
              },
            ],
            "id": 1,
            "link": "https://inews-article.com/test-article",
            "title": Object {
              "rendered": "title",
            },
          }
        }
        showAuthor={false}
      />
      <div
        className="inews__post-puff__content-headline"
      >
        <PostItemLabel
          post={
            Object {
              "breadcrumbs": Array [
                Object {
                  "name": "News",
                },
              ],
              "id": 1,
              "link": "https://inews-article.com/test-article",
              "title": Object {
                "rendered": "title",
              },
            }
          }
        />
        <PostItemTitle
          post={
            Object {
              "breadcrumbs": Array [
                Object {
                  "name": "News",
                },
              ],
              "id": 1,
              "link": "https://inews-article.com/test-article",
              "title": Object {
                "rendered": "title",
              },
            }
          }
        />
      </div>
    </div>
    <PostItemPubDate
      post={
        Object {
          "breadcrumbs": Array [
            Object {
              "name": "News",
            },
          ],
          "id": 1,
          "link": "https://inews-article.com/test-article",
          "title": Object {
            "rendered": "title",
          },
        }
      }
      showPubDate={false}
    />
  </div>
</div>
`;
